src/os/bluestore/BitAllocator.cc

```diff
@@ -707,7 +707,7 @@ void BitMapAreaIN::init(int64_t total_blocks, int64_t area_idx, bool def)
   BitMapArea **children = new BitMapArea*[num_child];
   int i = 0;
   for (i = 0; i < num_child - 1; i++) {
-    if (m_level <= 2) {
+    if (m_level < 2) {
```

Comment:
@chhabaramesh As per your own coding logic, if m_level == 2, you are going to call:
```
    if (m_level <= 2) {
      children[i] = new BitMapAreaLeaf(m_child_size_blocks, i, def);
```

And I check the initialization of BitMapAreaLeaf:
```c++
BitMapAreaLeaf::BitMapAreaLeaf(int64_t total_blocks, int64_t area_idx, bool def)
{
  init(total_blocks, area_idx, def);
}

void BitMapAreaLeaf::init(int64_t total_blocks, int64_t area_idx,
          bool def)
{
  int64_t num_child = 0;
  debug_assert(!(total_blocks % BmapEntry::size()));

  init_common(total_blocks, area_idx, def);
  num_child = total_blocks / pow(BitMapArea::get_span_size(), m_level);
  m_child_size_blocks = total_blocks / num_child;

  debug_assert(m_level == 1);

```

If we include the m_level == 2 case, isn't the last `debug_assert(m_level == 1);`  going to fire?